### DISCUSSION TITLE
Prevent reporting exceptions caused by user input.

### DISCUSSION THREAD

**Discussion by daniel-de-wit** [webonyx/graphql-php#1661] [Q&A] [answered]
How do I prevent an Exception caused by user input to be reported?

For example: the following GraphQL query throws an `GraphQL\Error\Error` exception when a user provides a number that exceeds the maximum Int value.

```GraphQL
query {
  example(
    # Int cannot represent non 32-bit signed integer value: 500000100000
    argument: 500000100000
  )
}
```

GraphQL\Error\Error
Int cannot represent non 32-bit signed integer value: 500000100000

----

I fear adding the exception to the $dontReport section will cause all errors from being reported.

```php
# app/Exceptions/Handler.php

class Handler extends ExceptionHandler
{
    /**
     * A list of the exception types that are not reported.
     *
     * @var array
     */
    protected $dontReport = [
        \GraphQL\Error\Error::class,
    ];
```

**Comment by spawnia** [comment, marked as answer]
Are you using Lighthouse?

Lighthouse will catch exceptions thrown during the execution of a query and return them as part of the response.

The error handler `Nuwave\Lighthouse\Execution\ReportingErrorHandler` is included in the default configuration
and reports non-client-safe errors through the default Laravel exception handler.

Client-safe errors are assumed to be something that:
- a client can understand and handle
- were caused by client misuse, e.g. wrong syntax, authentication, validation
Thus, they are typically not actionable for server developers.

**Reply by daniel-de-wit** [reply]
Yes I am using Lighthouse.

The way you describe how Lighthouse exception handler works is perfect.

I found the culprit to be a combination of 2 `.env` variables being set on the project.

```.env
APP_DEBUG=true
LIGHTHOUSE_DEBUG=15
```

After setting the default `LIGHTHOUSE_DEBUG=3` everything works as expected.

Thank you very much!
